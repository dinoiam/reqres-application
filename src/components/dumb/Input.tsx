import React, { InputHTMLAttributes } from 'react';
import styled from 'styled-components';

const InputWrapper = styled.div`
  position: relative;
  padding-top: 15px;
`;

const StyledInput = styled.input`
  border-radius: 5px;
  margin-bottom: 10px;
  padding: 10px;
  width: 100%;

  &.input-error {
    border-color: red;
  }
`;

const Error = styled.p`
  position: absolute;
  top: 0;
  color: red;
  font-size: 14px;
`;

type Props = InputHTMLAttributes<HTMLInputElement> & {
  /** boolean value which indicates if the input element is on error state */
  isOnError?: boolean;
  /** message shown if the input element is on error state */
  errorMessage?: string;
};

export const Input = ({
  isOnError,
  errorMessage,
  className,
  ...inputProps
}: Props): JSX.Element => {
  return (
    <InputWrapper>
      {isOnError && <Error>{errorMessage}</Error>}
      <StyledInput
        className={`${className} ${isOnError ? 'input-error' : ''}`}
        {...inputProps}
      ></StyledInput>
    </InputWrapper>
  );
};
