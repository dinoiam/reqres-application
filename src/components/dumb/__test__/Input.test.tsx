import React from 'react';
import { render } from '@testing-library/react';
import { Input } from '../Input';

describe('Input', () => {
  test('snapshot', () => {
    const { container } = render(<Input></Input>);

    expect(container).toMatchSnapshot();
  });
});
