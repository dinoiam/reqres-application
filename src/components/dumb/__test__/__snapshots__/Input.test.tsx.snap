// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input snapshot 1`] = `
<div>
  <div
    class="sc-bdfBQB ljxwmc"
  >
    <input
      class="sc-gsTEea bUaUCC undefined "
    />
  </div>
</div>
`;
